libsupport/include/katana/Random.h

```diff
@@ -8,15 +8,27 @@
 
 namespace katana {
 
-KATANA_EXPORT std::string RandomAlphanumericString(uint64_t len);
-// Range 0..len-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t len);
-// Range min+1..max-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t min, int64_t max);
-// Range 0.0f..max, inclusive
-KATANA_EXPORT float RandomUniformFloat(float max);
-
-KATANA_EXPORT std::mt19937& GetGenerator();
+using RandGenerator = std::mt19937;
+
+/// Generate a random alphanumeric string of length \param len using
+/// \param gen if provided. If no generator is specified, uses a generator
+/// seeded by the OS's randomness
+KATANA_EXPORT std::string RandomAlphanumericString(
+    uint64_t len, RandGenerator* gen = nullptr);
+
+/// Range 0..len-1, inclusive
+KATANA_EXPORT int64_t
+RandomUniformInt(int64_t len, RandGenerator* gen = nullptr);
```

Comment:
I don't know how I feel about this :D 

- These generators go to seed per-thread state so it is easy to not use this interface correctly
- Re: `GetGenerator(uint64_t)` part of the reason for taking over generator creation is that a `mt19937` has 19937 bits of state so seeding with an `int64_t` doesn't have a ton of entropy

Could you give me a sense of why you need this functionality?

We already have `RandGenerator& GetGenerator`. Could clients just reset that to whatever they want with some documentation about how this is per-thread?

If the ask is more specific (deterministic mode, testing only), there also might be a cleaner interface.

Reply:
The ask is a deterministic mode for testing. I want optional determinism for some data that I'm generating on the fly while leaving the random behavior intact to keep things like storage collisions from happening.

Reply:
I see.

I suppose there is some consistency benefit with using `RandomUniformInt` over `std::uniform_int_distribution` and `RandomAlphanumericString` doesn't have a standard analogue.

Do we really need `GetGenerator(uint64_t)` though? It's not really the right way to initialize a `std::mt19937`. It is fine to me if users want to opt into taking some shortcuts, but I don't think libsupport should imply that this is a common or equivalent way to initialize a generator compared to the default `GetGenerator()`)

Reply:
Word. Yeah; I'll ditch `GetGenerator(uint64_t)` and users that care can bring their own.